stglib/core/utils.py

```diff
@@ -937,6 +939,83 @@ def create_z(ds):
     return ds
 
 
+def create_z_bindist(ds):
```

Comment:
There is a lot of duplicate code here that appears to be from `create_z()`. Is there any way we could add `bindist` as an optional argument to `create_z()` instead of this additional function?